frontend/src/components/ShootWorkers/ManageWorkers.vue

```diff
@@ -120,6 +120,9 @@ export default {
       return findFreeNetworks(this.currentZonesNetworkConfiguration, this.existingWorkerCIDR, this.cloudProviderKind, this.allZones.length)
     },
     availableZones () {
+      if (!this.zonedCluster) {
```

Comment:
One last question before we merge your PR: Why did you add this code here? While I don't think that it will cause any problem with other infrastructures, I'm curious why you added it in the first place. I briefly tested your implementation and for me it worked without errors even without this code (at least not on the create cluster screen). Did you experience an error when changing an existing worker or did you just add this to be on the safe side?

Reply:
Because the hetzner provider implemenation was first without zones and the frontend assumes zones so i fixed some issues in this regard. We later switched to zones and the fixes became obsolete. I think they are still useful so I let them in this PR.

Reply:
Just for clarification: When did the issue occur? After shoot creation on the configure workers popup? Or somewhere else?

Reply:
When creating a now shoot, it complained about missing zones even the api returned no zones. I added the zonedCluster to return false in hcloud case and with this fix the problem was gone.